### ISSUE TITLE
[🕹️] Filter document analytics by link and views

### ISSUE THREAD

**Issue by mfts** [mfts/papermark#1228]
### What side quest or challenge are you solving?

In each document, we have aggregated document analytics and in order to better understand the analytics, we want to filter it. 

- filter by link
- filter by view
- exclude link(s)
- exclude view(s) 
- filter by viewer (i.e. email)

![CleanShot 2024-10-22 at 20 27 46@2x](https://github.com/user-attachments/assets/b9c6982f-fea7-453d-b2c8-0c2af2072236)

## Considerations

Please use url query params or the url query framework `nuqs` for controlling the state in the url for each filter. This way we always know what we are filtering on. 

We already have some functions and pipes in Tinybird that retrieve the relevant data for a set of viewIds 

https://github.com/mfts/papermark/blob/832b432e234b52f1c42d29f5afa92ff9db48256b/lib/tinybird/pipes.ts#L64-L73 

and the corresponding Tinybird pipe: 
https://github.com/mfts/papermark/blob/832b432e234b52f1c42d29f5afa92ff9db48256b/lib/tinybird/endpoints/get_document_duration_per_viewer.pipe

## Inspiration: 
I'm thinking about google flight search where the `airlines` filter has a toggle for **all**, **only one**, or **checkbox to (de)select multiple airlines** 

![CleanShot 2024-10-22 at 20 32 33@2x](https://github.com/user-attachments/assets/61e198b9-8384-425e-ad90-264809927003)

--- 
Please only serious PRs. This will likely take some experimenting, you need to create a tinybird account in order to fully test the implementation yourself.

### Points

1500 Points

### Provide proof that you've completed the task

Open a PR with working code to the solution, detailed explanation what you did and ideally a video of the working solution.

**Comment by RajuGangitla** [comment]
@mfts im trying to setup the tiny bbird database and it added bunch of files 

![image](https://github.com/user-attachments/assets/ae0e3dad-4148-45b1-ba7f-fe144e440b69)


![image](https://github.com/user-attachments/assets/c2bddbfd-572e-49cf-b014-0463a1a80731)

**Comment by mfts** [comment]
@RajuGangitla Yes that's adding all the python files. You don't need to commit them. 

Did you open open a python shell with pipenv?

**Comment by mfts** [comment]
@RajuGangitla If you have pipenv install on your computer, then inside the repo directory, you can call `pipenv shell` to open a python shell with the tinybird package installed.

**Comment by RajuGangitla** [comment]
@mfts Its stilling  Error: invalid authentication token. Workspace not found, make sure you use the token host https://api.us-east.tinybird.co 

https://github.com/user-attachments/assets/544b8051-5e75-4f1b-8bf3-58c2254c1e47

**Comment by mfts** [comment]
In tinybird when you create the account and token, you need to add the pipe to the token permissions. You do that inside the tinybird dashboard

**Comment by RajuGangitla** [comment]
maybe that time i didnt add that 
is there any way to add pipes to existing tokesns 
if not  i need to create a new token ?
![image](https://github.com/user-attachments/assets/ab76528c-f2c1-4943-88d6-31c2a430572b)


is this the pipes i need to add 
![image](https://github.com/user-attachments/assets/63054a39-594c-4d4c-9edc-f2f033ea1cc3)

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by DeepaPrasanna** [comment]
hey @RajuGangitla , if you are not working on this, may I take this forward?

**Comment by RajuGangitla** [comment]
@DeepaPrasanna I'm working on this one

**Comment by mfts** [comment]
@RajuGangitla let me know if you need help :)

**Comment by RajuGangitla** [comment]
@mfts even I have added token it's still saying invalid authentication token do you have any idea

**Comment by RajuGangitla** [comment]
@mfts I'm interested to work on this one 

But I didn't understand why I'm getting invalid authentication token for tinybird token 

1) I did authentication through cli
It's successful 
2) I push the datasources and data pipes 
3) now if I'm trying to get the data or ingest the data it's throwing error 

I added tinybird token in env

**Comment by TSP-Dev** [comment]
got same error.
